File: app/src/main/java/com/agrogames/islandsofwar/engine/abs/another/IGraphicsObject.java
package com.agrogames.islandsofwar.engine.abs.another;

import com.agrogames.islandsofwar.engine.abs.renderable.RenderableObject;
import com.agrogames.islandsofwar.engine.abs.updatable.UpdatableObject;

public interface IGraphicsObject extends RenderableObject, UpdatableObject {
    boolean shouldBeRemoved();
}

File: app/src/main/java/ru/agrogames/islands/engine/abs/graphics/IGraphicsObject.kt
package ru.agrogames.islands.engine.abs.graphics

import ru.agrogames.islands.engine.abs.renderable.RenderableObject
import ru.agrogames.islands.engine.abs.updatable.UpdatableObject

interface IGraphicsObject : RenderableObject, UpdatableObject {
    fun shouldBeRemoved(): Boolean
    val spawnSound: String?
}